src/js/utils/helpers.ts

```diff
@@ -55,17 +56,19 @@ function crossdomain(uri) {
 }
 
 // The predicate received the arguments (key, value) instead of (value, key)
-const foreach = function (aData, fnEach) {
+const foreach = function (aData: GenericObject, fnEach: (key: string, value: any) => void): void {
     for (let key in aData) {
         if (Object.prototype.hasOwnProperty.call(aData, key)) {
             fnEach(key, aData[key]);
         }
     }
 };
 
-const noop = function () {};
+const noop = () => {
+    // noop
+};
 
-const helpers = Object.assign({}, parser, validator, playerutils, {
+const helpers: GenericObject = Object.assign({}, parser, validator, playerutils, {
```

Comment:
It might be interesting to type this as it goes on the public api as `jwplayer().utils` and in commercial for backward compatibility `jwplayer.utils`. See https://github.com/jwplayer/jwplayer/blob/master/test/unit/utils-defs-test.js

Reply:
Good call, will update

Reply:
As per earlier discussion, I'd like to wait on this until the dependencies are typed, since that will have a similar result.